src/Sarif.Multitool/AbsoluteUriCommand.cs

```diff
@@ -23,24 +23,34 @@ public int Run(AbsoluteUriOptions absoluteUriOptions)
         {
             try
             {
-                var sarifFiles = GetSarifFiles(absoluteUriOptions);
+                IEnumerable<AbsoluteUriFile> absoluteUriFiles = GetAbsoluteUriFiles(absoluteUriOptions);
 
-                if (!absoluteUriOptions.Inline)
+                bool outputFilesCanBeCreated = true;
+                foreach (AbsoluteUriFile absoluteUriFile in absoluteUriFiles)
                 {
-                    Directory.CreateDirectory(absoluteUriOptions.OutputFolderPath);
+                    outputFilesCanBeCreated &=
+                        DriverUtilities.ReportWhetherOutputFileCanBeCreated(
+                            absoluteUriFile.OutputFilePath,
+                            absoluteUriOptions.Force,
+                            _fileSystem);
                 }
 
-                foreach (var sarifLog in sarifFiles)
+                if (!outputFilesCanBeCreated) { return 1; }
```

Comment:
>1 [](start = 55, length = 1)

Do we use the return codes for anything. i.e. should we return a different error code than the unhandled exception path?

Reply:
We only check 0/non-zero. No need at this time for distinguished non-zero codes.

---
In reply to: [312257820](https://github.com/microsoft/sarif-sdk/pull/1625#discussion_r312257820) [](ancestors = 312257820)